tests/cupy_tests/functional_tests/test_vectorize.py

```diff
@@ -85,7 +84,8 @@ def run_div(self, func, xp, dtypes):
         return f(x1, x2)
 
     @testing.for_all_dtypes_combination(names=('dtype1', 'dtype2'))
-    @testing.numpy_cupy_allclose(rtol=1e-6)
+    # TODO(asi1024): Fix return type.
+    @testing.numpy_cupy_allclose(rtol=1e-3, type_check=False)
```

Comment:
```py
>>> numpy.vectorize(lambda x, y: x / y)(numpy.array([1], dtype=numpy.uint16), numpy.array([1], dtype=numpy.float32)).dtype
dtype('float64')
```

Reply:
@seberg Question: I found that the dtype of the result of true_divide of int16 scalar and float32 scalar in numpy is different between NumPy 1.24 and earlier. Is this an intentional change or a bug?

NumPy 1.23.5
```py
>>> numpy.__version__
'1.23.5'
>>> (numpy.int16(1) + numpy.float32(1)).dtype
dtype('float32')
>>> (numpy.int16(1) / numpy.float32(1)).dtype
dtype('float32')
```

NumPy 1.24.1
```py
>>> numpy.__version__
'1.24.1'
>>> (numpy.int16(1) + numpy.float32(1)).dtype
dtype('float32')
>>> (numpy.int16(1) / numpy.float32(1)).dtype
dtype('float64')
```

Reply:
Arrg, no that is a subtle bug in my new scalar logic :(.  The `int / int -> float64`, so it tries converting float32 to that and succeeds, but since the other is a float64, it must not do that.

Reply:
xref: https://github.com/numpy/numpy/pull/23079

cc @leofang (for awareness)

Reply:
Thanks Sebastian for fixing this quickly!

@asi1024 Let's mark the test skipif numpy==1.24.0 or 1.24.1.